### ISSUE TITLE
IMGUI on Apple Silicon M1 support

### ISSUE THREAD

**Issue by vis2k** [dotnet/Silk.NET#894]
# Summary
Building the imgui example on M1 currently throws an exception about missing 'cimgui' library:
```
Unhandled exception. System.IO.FileNotFoundException: Could not find or load the native library: cimgui.dylib Attempted: "cimgui.dylib", "[MY PATH]/bin/Debug/net6.0/cimgui.dylib"
File name: 'cimgui.dylib'
   at Silk.NET.Core.Loader.LibraryLoader.ThrowLibNotFound(String name, PathResolver resolver)
   at Silk.NET.Core.Loader.LibraryLoader.LoadNativeLibrary(String name, PathResolver pathResolver)
   at Silk.NET.Core.Loader.LibraryLoader.LoadNativeLibrary(String name)
   at Program.Main(String[] args) in [MY PATH]/Program.cs:line 28

```

Nothing shows up in google about this, so I'll gather what I figured out so far.
quotes from discord, mostly roeyskoe.

- the library isn't found "because the version of Imgui.NET silk.NET uses"
- "There is a newer version of Imgui.NET out "with M1 support", but it is broken."
- according to , this is a ImGUI.NET
- manually copying the cimgui.dylib from here: https://github.com/mellinoe/ImGui.NET/blob/master/deps/cimgui/osx-universal/cimgui.dylib to bin/Debug/net6.0/ opens the Mac security popup when running the sample
- in mac security settings, need to allow running the library
- running again still crashes the project though

supposedly another tweak needed: "That was with one other silk tweak, since the dylib is in a place where LibraryLoader does not try to find it"

link to discord conversations:

  https://discord.com/channels/521092042781229087/607634593201520651/965976744244961310
  https://discord.com/channels/521092042781229087/587346162802229298/959064581899190323

**Comment by vis2k** [comment]
https://github.com/mellinoe/ImGui.NET/pull/318 :)))

**Comment by vis2k** [comment]
btw I found this commit in ImGUI.NET regarding apple silicon:
https://github.com/mellinoe/ImGui.NET/commit/ae3ecd014e416a864cc49ec361ad8f903e22656d

**Comment by vis2k** [comment]
history of cimgui.dylib builds for IMGUI.NET here:
https://github.com/mellinoe/ImGui.NET-nativebuild/releases

**Comment by vis2k** [comment]
Silk.Net.Opengl.Extensions.Imgui v2.15.0 uses IMGUI.NET 1.86.0..

**Comment by vis2k** [comment]
> history of cimgui.dylib builds for IMGUI.NET here: https://github.com/mellinoe/ImGui.NET-nativebuild/releases

trying the 1.86.0 cimgui.dylib. renamed to libcimgui. finds the file but wrong architecture:
"(mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e'))"

**Comment by vis2k** [comment]
imgui.net-nativebuild 1.86.0 uses cimgui at this exact commit:
https://github.com/cimgui/cimgui/commit/49d45008e046f0123d464bd323d16319c588b8cc

gonna try to build it myself

**Comment by vis2k** [comment]
managed to build cimgui.dylib myself on apple silicon:
<img width="547" alt="2022-04-20_15-25-38@2x" src="https://user-images.githubusercontent.com/16416509/164184926-9469eed3-b7fe-4f9f-9906-356d1dcfa374.png">

**Comment by vis2k** [comment]
success:
![2022-04-20_15-27-22@2x](https://user-images.githubusercontent.com/16416509/164185296-df5de13b-7f2d-4b7f-b37a-6bfe8472d98f.png)

**Comment by vis2k** [comment]
**instructions to use Silk.Net 2.15's imgui.net on apple silicon M1:**

**option 1:** put libcimgui file into YOUR_PROJECT/bin/Debug/net6.0/libcimgui  (no file extension)
file is uploaded in discord:
https://discord.com/channels/521092042781229087/607634593201520651/966254892580749313

**option 2:** build it yourself:
- download this folder: https://github.com/mellinoe/ImGui.NET-nativebuild/tree/4d1764caa3e05818f3d85d84d6f0dd05afb5bbf0
- the cimgui folder is empty. put this exact version into it: https://github.com/cimgui/cimgui/tree/49d45008e046f0123d464bd323d16319c588b8cc
- the imgui folder inside the cimgui folder is empty. put this exact version into it: https://github.com/ocornut/imgui/tree/15b4a064f9244c430e65214f7249b615fb394321
- then go back tot he root folder and run "./build-native.sh"
- the file will be in /cimgui/build/Debug/cimgui.dylib
- copy this YOUR_PROJECT/bin/Debug/net6.0/libcimgui (rename it and remove the file extension)

done.

**Comment by Perksey** [comment]
Removed from 2.X milestone, this isn't a Silk.NET bug.

**Comment by Perksey** [comment]
Actually I'm just going to close this, there's a PR upstream and until that's merged (next decade) there's nothing we can do here.

**Comment by vis2k** [comment]
> Actually I'm just going to close this, there's a PR upstream and until that's merged (next decade) there's nothing we can do here.

Maybe useful to keep it open for apple silicon users to find it, including the workaround?
As for Silk.NET, you guys could include the cimgui file manually like I explained above.
This way, Silk.NET IMGUI would actually run on apple silicon :)

**Comment by Perksey** [comment]
ImGui bindings are not in scope for Silk.NET today and, as a result, neither are shipping native packages. Given that there's nothing we can do here to fix it in scope with the project and it's not a specific part within our packages that are blocked by this upstream bug, it doesn't warrant an issue being kept open here.
